frontend/pulse3d/src/components/uploadForm/AnalysisParamForm.js

```diff
@@ -349,12 +352,26 @@ export default function AnalysisParamForm({
             <DropDownWidget
               options={pulse3dVersions.map((version) => {
                 const selectedVersionMeta = metaPulse3dVersions.filter((meta) => meta.version === version);
-                return selectedVersionMeta[0] && selectedVersionMeta[0].state === "testing"
-                  ? version + " " + "[ testing ]"
-                  : version;
+                if (selectedVersionMeta[0] && selectedVersionMeta[0].state === "testing") {
+                  return version + " " + "[ testing ]";
+                } else if (selectedVersionMeta[0] && selectedVersionMeta[0].state === "deprecated") {
+                  return version + " " + "[ deprecated ]";
+                } else {
+                  return version;
+                }
               })}
               reset={!checkedParams}
               handleSelection={(idx) => {
+                const selectedVersionMetaData = metaPulse3dVersions.filter(
```

Comment:
I'd move these functions for `handleSelection` and `options` out of the html to make it more readable